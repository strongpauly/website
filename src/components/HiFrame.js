import React, { Component } from 'react';

import PropTypes from 'prop-types';

export default class HiFrame extends Component {
  static propTypes = {
    src: PropTypes.string,
    title: PropTypes.string
  }

  render() {
    //Resizes the iframe to it's content, then sets focus.
    function fixFrame(e) {
      const iframe = e.target;
      iframe.style.height = iframe.contentDocument.body.scrollHeight + 'px';
      iframe.contentWindow.focus();
    }
    return <iframe src={this.props.src} title={this.props.title} onLoad={fixFrame}></iframe>;
  }

}
