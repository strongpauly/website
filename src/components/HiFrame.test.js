import React from 'react';
import {HiFrame} from './HiFrame';
import {shallow} from 'enzyme';

/* eslint-env jest */

describe('HiFrame', () => {
  it('renders without crashing', () => {
    const app = shallow(<HiFrame src="somewhere" title="something"/>);
    expect(app).toMatchSnapshot();
  });

  it('will resize frame onLoad', () => {
    const app = shallow(<HiFrame src="somewhere" title="something"/>);
    const iframe = app.find('iframe');
    expect(iframe).toHaveLength(1);
    const focus = jest.fn();
    const mock = {style:{}, contentDocument: {body: {scrollHeight: 8}}, contentWindow:{focus:focus}};
    iframe.simulate('load', {target: mock});
    expect(mock.style.height).toEqual('8px');
    expect(focus).toHaveBeenCalled();

  });
});
