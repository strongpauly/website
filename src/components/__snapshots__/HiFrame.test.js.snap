// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HiFrame renders without crashing 1`] = `
<iframe
  onLoad={[Function]}
  src="somewhere"
  title="something"
/>
`;
